pkgs/development/compilers/openjdk/corretto.nix

```diff
@@ -0,0 +1,88 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, fetchpatch
+, bash
+, openjdk8
+
+, ...
+} @ args:
+
+let
+  version = "8.202.08.2";
+  src = fetchFromGitHub {
+    owner  = "corretto";
+    repo   = "corretto-8";
+    rev    = "1cb3ca2b6f0d9466bb982e31ee64ca138422f0fe";
+    sha256 = "1wxbfiv5gfyy4mimqy8zwa0xa46pb1xk45a2z6w5zsnrkam2c74a";
+  };
+
+  name = "corretto-${version}";
+
+  # In the form: $MAJOR.$UPDATE.$BUILD.$REVISION
+  tokens = lib.splitString "." version;
+
+  majorVersion     = builtins.elemAt tokens 0;
+  updateVersion    = builtins.elemAt tokens 1;
+  buildNumber      = builtins.elemAt tokens 2;
+  correttoRevision = builtins.elemAt tokens 3;
+
+  corretto = openjdk8.overrideAttrs (oldAttrs: rec {
+    inherit name version src;
+
+    # Override upstream sources and directory auto-detection; Corretto includes
+    # everything in the same repository.
+    srcs = null;
+    sourceRoot = null;
+
+    # OpenJDK sources in the repo are stored under './src'
+    postUnpack = ''
+      sourceRoot+=/src
+    '' + (oldAttrs.postUnpack or "");
+
+    # Don't need the 'move everything to the same directory' steps that the main
+    # recipe does.
+    prePatch = null;
+
+    patches = oldAttrs.patches ++ [
+      # Suppress an error from -Wformat
+      (fetchpatch {
+        url = "http://icedtea.classpath.org/hg/icedtea8/raw-file/cea1a48810dd/patches/pr3597.patch";
+        sha256 = "1dy79l5jq8npdx35i9aq9xai5lz9k35db65phkvham69ch4564g7";
+      })
+
+      # Some minor patches to OpenJDK to get everything building; taken from:
+      #   https://github.com/corretto/corretto-8/issues/23
+      ./corretto.patch
```

Comment:
Is it possible to fetch rather than add the patch to the repo?

Reply:
The actual patch isn't a PR on the main repository, but rather a commit from a fork:
https://github.com/dacut/corretto-8/commit/68c43e760ff22dfd94ab6f2c05cea3d53190439a

I thought that committing the patch would be more resilient to the fork being deleted or changed, but happy to `fetchpatch` that diff if you'd rather.

Reply:
By itself this is benign, I guess, but it still adds to the overall amount of unnecessary stuff in the repo. Your call.

Reply:
(Source files are normally mirrored by the nixos org, can't say whether patch files are without checking the script)

Reply:
I think I'll keep it; if it was more than 2 lines, I'd fetch it, but as-is I think it's pretty small.